### PR TITLE
Icon grid component

### PR DESCRIPTION
**Pull request by bennpearson**
### What is the context of this PR?
Added icon grid component

### How to review 
Check component matches designs: https://docs.google.com/presentation/d/1D80cOrhRs-p-jCE3nZ1UHPkSyZQVOIeyjpB-5rSzVqA/edit#slide=id.g248faee3f0_1_4

### CODE REVIEW

**Comment by bennpearson** [comment]
[Icon Grid Component (3)](https://trello.com/c/VstsKMdc/126-icon-grid-component-3)

File: src/components/IconGrid/IconGridButton.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    );
+  });
+
+  it("will render an icon", function() {
+    expect(wrapper.find("img").props().src).toContain(icons.checkbox);
+  });
+
+  it("will render a title", function() {
+    expect(wrapper.props().title).toEqual("Checkbox");
+  });
```
Comment (WickyNilliams) [inline src/components/IconGrid/IconGridButton.test.js]: I'd use a snapshot test for anything render related

File: src/components/IconGrid/story.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <IconGrid>
+    <IconGridButton icon="checkbox" title="checkbox" />
+    <IconGridButton icon="currency" title="currency" />
+    <IconGridButton icon="date" title="date" />
+    <IconGridButton icon="number" title="number" />
+    <IconGridButton icon="radio" title="radio" />
+    <IconGridButton icon="select" title="select" />
+    <IconGridButton icon="textarea" title="textarea" />
+    <IconGridButton icon="textfield" title="textfield" />
+    <IconGridButton icon="time" title="time" />
```
Comment (WickyNilliams) [inline src/components/IconGrid/story.js]: Have a look at using storybook's `action` addon for logging callbacks etc. It's used already in `CanvasSection.story.js`

File: src/components/IconGrid/IconGrid.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      {children}
+    </Menu>
+  );
+};
+
+IconGrid.propTypes = {
+  children: PropTypes.node.isRequired
+};
+
+export default IconGrid;
```
Comment (WickyNilliams) [inline src/components/IconGrid/IconGrid.js]: No tests for this component?

File: src/components/IconGrid/IconGridButton.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  margin: 1em;
+  padding: 0;
+  cursor: pointer;
+  background: transparent;
+  border: none;
+  opacity: 1;
+  transition: all 200ms ease-in-out;
+  &:hover,
+  &:focus {
+    background-color: #eaeaea;
```
Comment (WickyNilliams) [inline src/components/IconGrid/IconGridButton.js]: we already have this color in `theme.js`, worth using from there?

File: src/components/IconGrid/IconGridButton.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      </Title>
+    </Button>
+  );
+};
+
+IconGridButton.propTypes = {
+  icon: PropTypes.oneOf(keys(icons)),
+  title: PropTypes.string.isRequired,
+  disabled: PropTypes.bool,
+  handleClick: PropTypes.func
```
Comment (WickyNilliams) [inline src/components/IconGrid/IconGridButton.js]: call this `onClick` (surprised eslint hasn't complained about this!)

**Pull request merged**